### PR TITLE
Only set INSTALLATION environment variables when doing injection

### PR DESCRIPTION
**Pull request by stanistan**
### What does this PR do?

https://datadoghq.atlassian.net/browse/APMS-14061

We no longer set the environment variables when there is no library to inject (disabled namespace or otherwise).

### Motivation

https://docs.google.com/document/d/1PHDp652AmVmNG--cJf0JpV_5OB6evtHZ_qxoZI2z0Pk/edit?tab=t.0

### Describe how to test/QA your changes

Tests updated.

### CODE REVIEW

**Review by paullegranddc — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

File: pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go [hunk 1]
```diff
@@ -227,26 +226,6 @@ func profilingClientLibraryConfigMutators(config *webhookConfig) []podMutator {
 	return podMutators
 }
 
-func injectApmTelemetryConfig(pod *corev1.Pod) {
```
Comment (adel121) [inline pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go:230]: question: why did you move this from a function call to an inline implementation?

**Review by adel121 — COMMENTED** [review]
left small question

File: pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go (hunk 1 above)
Comment (stanistan) [reply pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go:230]: The inline implementation is in a `podMutator` which is basically exactly what this is. It didn't seem necessary to keep this around.

Comment (stanistan) [reply pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go:230]: I could have just called this function there I guess. 🤷

Comment (stanistan) [reply pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go]: @adel121 updated to use the function.

Comment (adel121) [reply pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go]: Thanks 👍

**Review by adel121 — APPROVED** [review]

**Review by iamluc — APPROVED** [review]

File: releasenotes-dca/notes/admission-controller-disabled-no-env-vars-135252f0eb8ddef4.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    The autoinstrumentation webhook no longer injects default environment
```
Comment (rtrieu) [inline releasenotes-dca/notes/admission-controller-disabled-no-env-vars-135252f0eb8ddef4.yaml:4]:
```suggestion
    The auto-instrumentation webhook no longer injects the default environment
```

**Review by rtrieu — APPROVED** [review]
Approving with minor feedback.
